shell/app/modules/application/pages/pipeline/run-detail/build-log.tsx

```diff
@@ -131,23 +135,50 @@ export class PureBuildLog extends React.PureComponent<IProps, IState> {
         onChange={this.toggleLogName}
       />
     );
-    if (withoutDrawer) {
-      return (
-        <CompSwitcher comps={slidePanelComps}>
-          <LogRoller
-            key={String(isStdErr)}
-            query={{
-              fetchApi: customFetchAPIPrefix || `/api/cicd/${pipelineID}/tasks/${taskID}/logs`,
-              downloadAPI,
-              taskID,
-              stream: isStdErr ? 'stderr' : 'stdout',
-            }}
-            logKey={logId}
-            extraButton={switchLog}
-            transformContent={this.transformContentToLink}
-          />
-        </CompSwitcher>
+
+    const logRollerComp =
+      taskContainers.length !== 0 ? (
+        <Tabs defaultActiveKey="1">
+          {taskContainers.map((item) => (
+            <TabPane tab={item.taskName} key={item.containerID}>
+              <LogRoller
+                key={String(isStdErr)}
+                query={{
+                  // 此处如果使用了customFetchAPIPrefix，在drawer关闭瞬间又触发了请求会使用后面的api，此时pipelineID为undefined(见自动化测试)
+                  fetchApi:
+                    customFetchAPIPrefix ||
+                    (pipelineID && taskID ? `/api/cicd/${pipelineID}/tasks/${taskID}/logs` : false),
+                  downloadAPI,
+                  taskID,
+                  id: item.containerID,
```

Comment:
these two log comp definition are almost same， try to extract and combine into one